<launch>

  <!-- Publicar odometria y tf -->
  <node name="odom_publisher" pkg="bring_up" type="fake_odom_publisher.py" output="screen"/>

  <!-- Add Rviz-->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find bring_up)/rviz/QT1.rviz"/>

  <!-- Robot description (ilhuacamina)-->
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find bring_up)/URDF/ilhuicamina_simplified.xacro'"/>
  <node name="robot_spawn" pkg="gazebo_ros" type="spawn_model" output="screen"
        args="-urdf -param robot_description -model QT1 "/>
        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
                <param name="use_tf_static" value="false"/>
  </node>

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/>


  <!-- Adding wheel tf broadcaster -->
  <node name="wheel_tf_broadcaster" pkg="bring_up" type="wheel_tf_broadcaster.py" output="screen"/>


  <!-- Add gmapping for SLAM mapping-->
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" clear_params="true">
        <param name ="base_frame" value="base_link" />
        <param name="map_update_interval" value="30.0"/>
  </node>

</launch>